### PR TITLE
[isar_generator] Add successes test

### PR DESCRIPTION
**Pull request by blendthink**
## Overview

I created 2 types tests related to PR #496.

- type_not_specified
- name_annotation

However, this may not fit the development policy.
In that case, there is no problem to close this.

### CODE REVIEW

File: packages/isar/lib/src/native/binary_writer.dart [hunk 1]
```diff
@@ -219,7 +219,9 @@ class BinaryWriter {
   }
 
   void validate() {
-    assert(_dynamicOffset == _buffer.length,
-        'Invalid write buffer size. $_dynamicOffset != ${_buffer.length}');
+    assert(
+      _dynamicOffset == _buffer.length,
+      'Invalid write buffer size. $_dynamicOffset != ${_buffer.length}',
+    );
```
Comment (blendthink) [inline packages/isar/lib/src/native/binary_writer.dart:225]: In addition, I fixed lint error.
Please let me exclude Codecov warnings from this PR scope.

**Comment by leisim** [comment]
Thanks for this PR it is a good start. Ideally we would find a way to also verify that the output of the generator is correct.

**Comment by blendthink** [comment]
@leisim 
Thanks for commenting!

> Ideally we would find a way to also verify that the output of the generator is correct.
> But I don't know a good way how to do that. Do you have an idea?

This case was more of a test on success of `IsarAnalyzer().analyze(element)`.
If we would like to verify that the output of the generator is correct, we would be good to use [source_gen_test](https://pub.dev/packages/source_gen_test).


For example, it is used in `retrofit`.

https://github.com/trevorwang/retrofit.dart/blob/691c916da5b8a815aaead664ab704e6565810ee0/generator/test/src/generator_test_src.dart#L7-L18
https://github.com/trevorwang/retrofit.dart/blob/691c916da5b8a815aaead664ab704e6565810ee0/generator/test/generator_test.dart#L10-L16

I'll try to implement it now.
Please wait for a moment.

**Comment by blendthink** [comment]
I implemented. https://github.com/isar/isar/pull/501/commits/0a62cd45332116a680ec43f15f5e6f07f654c067

Something similar can be implemented with `build_test`, but it is more complicated.
```dart
// ...
await testBuilder(
  getIsarGenerator(BuilderOptions.empty),
  {'a|${file.path}': content.join('\n')},
  reader: await PackageAssetReader.currentIsolate(),
  outputs: {
    'a|${file.path.replaceFirst('.dart', '.isar_generator.g.part')}':
        decodedMatches(contains('xxx'))
  },
);
```

What do you think?

**Comment by blendthink** [comment]
@leisim 
I have a feeling that major changes are coming, so I'll close it for now to avoid unnecessary distractions.
If requested, I will incorporate the latest main branch changes and resubmit, so please contact me at any time.

**Pull request closed without merging**